rapids-cmake/test/generate_resource_spec.cmake

```diff
@@ -46,7 +46,19 @@ files to be generated.
 function(rapids_test_generate_resource_spec DESTINATION filepath)
   list(APPEND CMAKE_MESSAGE_CONTEXT "rapids.test.generate_resource_spec")
 
-  if(NOT DEFINED CMAKE_CUDA_COMPILER AND NOT DEFINED CMAKE_CXX_COMPILER)
+  set(rapids_have_cxx OFF)
+  set(rapids_have_cuda OFF)
+  get_property(rapids_languages GLOBAL PROPERTY ENABLED_LANGUAGES)
+  if("CXX" IN_LIST rapids_languages)
+    set(rapids_lang CXX)
+    set(rapids_have_cxx ON)
+  endif()
+  if("CUDA" IN_LIST rapids_languages)
+    set(rapids_lang CUDA)
+    set(rapids_have_cuda ON)
+  endif()
+
+  if(NOT (rapids_have_cxx OR rapids_have_cuda))
```

Comment:
Seems to me like you could get rid of `rapids_have_cxx` and `rapids_have_cuda` altogether, replace them with `unset(rapids_lang)`, and check for `rapids_lang` here. WDYT?

Reply:
That is great improvement. Updated